src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts

```diff
@@ -3167,208 +3167,647 @@ describe('SQL Builder test', () => {
+      computeMethod: ExploreComputeMethod.USER_CNT,
+      specifyJoinColumn: true,
+      joinColumn: 'user_pseudo_id',
+      conversionIntervalType: ExploreConversionIntervalType.CUSTOMIZE,
+      conversionIntervalInSeconds: 10*60,
+      eventAndConditions: [
+        {
+          eventName: 'add_button_click',
+          sqlCondition: {
+            conditions: [{
+              category: 'other',
+              property: 'platform',
+              operator: '=',
+              value: ['Android'],
+              dataType: MetadataValueType.STRING,
+            },
+            {
+              category: 'device',
+              property: 'screen_height',
+              operator: '<>',
+              value: [1400],
+              dataType: MetadataValueType.INTEGER,
+            }],
+            conditionOperator: 'and',
+          },
+        },
+        {
+          eventName: 'note_share',
+          sqlCondition: {
+            conditions: [{
+              category: 'other',
+              property: 'platform',
+              operator: '=',
+              value: ['Android'],
+              dataType: MetadataValueType.STRING,
+            },
+            {
+              category: 'device',
+              property: 'screen_height',
+              operator: '<>',
+              value: [1400],
+              dataType: MetadataValueType.INTEGER,
+            }],
+            conditionOperator: 'or',
+          },
+
+        },
+        {
+          eventName: 'note_export',
+        },
+      ],
+      timeScopeType: ExploreTimeScopeType.FIXED,
+      groupColumn: ExploreGroupColumn.DAY,
+      timeStart: new Date('2023-04-30'),
+      timeEnd: new Date('2023-06-30'),
+      pathAnalysis: {
+        sessionType: ExplorePathSessionDef.CUSTOMIZE,
+        nodeType: ExplorePathNodeType.EVENT,
+        lagSeconds: 3600,
+      },
+    });
+
+    const expectResult = `
+    with
+      tmp_data as (
+        select
+          event_date,
+          event_name,
+          event_id,
+          event_bundle_sequence_id::bigint as event_bundle_sequence_id,
+          event_previous_timestamp::bigint as event_previous_timestamp,
+          event_server_timestamp_offset::bigint as event_server_timestamp_offset,
+          event_timestamp::bigint as event_timestamp,
+          ingest_timestamp,
+          event_value_in_usd,
+          app_info.app_id::varchar as app_info_app_id,
+          app_info.id::varchar as app_info_package_id,
+          app_info.install_source::varchar as app_info_install_source,
+          app_info.version::varchar as app_info_version,
+          device.vendor_id::varchar as device_id,
+          device.mobile_brand_name::varchar as device_mobile_brand_name,
+          device.mobile_model_name::varchar as device_mobile_model_name,
+          device.manufacturer::varchar as device_manufacturer,
+          device.screen_width::bigint as device_screen_width,
+          device.screen_height::bigint as device_screen_height,
```

Comment:
Add tests for the common attribute `device.viewport_width` and `device.viewport_height` for the Web SDK